### ISSUE TITLE
Change _parse_nodes to improve recursing through a scenegraph

### ISSUE THREAD

**Issue by Mateasek** [mattngc/raysect_mayavi#4]
The _parse_nodes method now can't handle correctly any class inherited from the Node class and can't handle primitives being parents to other primitives, [see the if statement](https://github.com/mattngc/raysect_mayavi/blob/5519de7267b5fdb0b195a7ba71575f775c27a351/raysect_mayavi/scenegraph_viewer.py#L12-L17). I think a better solution could be:

```python
        if isinstance(child, Primitive):
            _parse_nodes(child, mesh_list)
            mesh_list.append((to_mesh(child), child.meta))
        elif isinstance(child, Observer):
            return
        elif isinstance(child, Node):
            _parse_nodes(child, mesh_list)
```
Also, I think it could be made a function returning a list of meshes instead of changing a list.

**Comment by mattngc** [comment]
Hi Matej, ok, I think I understand what you are suggesting. Happy to accept a merge request with this change.

**Comment by Mateasek** [comment]
Hi Matt,

made a PR #5 with suggested changes, at the end I went for a slightly different solution.

**Comment by mattngc** [comment]
Resolved with PR #5
